Mathlib/Algebra/Category/ModuleCat/Presheaf/ChangeOfRings.lean

```diff
@@ -0,0 +1,57 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.Algebra.Category.ModuleCat.ChangeOfRings
+import Mathlib.Algebra.Category.ModuleCat.Presheaf
+
+/-!
+# Change of presheaf of rings
+
+In this file, we define the restriction of scalars functor
+`restrictScalars α : PresheafOfModules.{v} R' ⥤ PresheafOfModules.{v} R`
+attached to a morphism of presheaves of rings `α : R ⟶ R'`.
+
+-/
+
+universe v v' u u'
+
+open CategoryTheory
+
+namespace PresheafOfModules
+
+variable {C : Type u'} [Category.{v'} C] {R R' : Cᵒᵖ ⥤ RingCat.{u}}
+
+/-- The restriction of scalars of presheaves of modules, on objects. -/
+@[simps]
+noncomputable def restrictScalarsBundledCore (M' : PresheafOfModules R') (α : R ⟶ R') :
+    BundledCorePresheafOfModules R where
+  obj X := (ModuleCat.restrictScalars (α.app X)).obj (M'.obj X)
+  map {X Y} f :=
+    { toFun := fun x ↦ M'.map f x
+      map_add' := by intros; simp only [map_add, map_apply]
+      map_smul' := fun r x ↦ by
+        have eq := RingHom.congr_fun (α.naturality f) r
+        dsimp at eq
+        apply (M'.map_smul f (α.app _ r) x).trans
+        dsimp
```

Comment:
```suggestion
```
This line doesn't seem to be strictly necessary, but maybe this is not a good idea because too much burden is placed on `rfl` and I have no idea how efficient that is.

Reply:
I have used the syntax `dsimp at eq ⊢` which also reduces the number of lines of code by `1`!